src/scales/scale.time.js

```diff
@@ -358,31 +358,67 @@ function generate(min, max, capacity, options) {
 }
 
 /**
- * Returns the right and left offsets from edges in the form of {left, right}.
+ * Returns the right and left offsets from the edges of the chart in the form of {left, right}
+ * where each values is a relative width to the scale and ranges between 0 and 1.
+ * They add extra margins on the both sides by scaling down the original scale.
+ * Offsets are typically used for bar charts. They are calculated based on intervals
+ * between the data points to keep the leftmost and rightmost bars from being cut.
  * Offsets are added when the `offset` option is true.
  */
-function computeOffsets(table, ticks, min, max, options) {
+function computeOffsets(table, ticks, data, min, max, options) {
+	var pos = [];
+	var minInterval = 1;
+	var timeOpts = options.time;
+	var barThickness = options.barThickness;
```

Comment:
Accessing `barThickness` from the time scale doesn't seem ideal because it correlates the bar controller and this scale, which is something we are trying to remove / avoid. Actually, the `barThickness` should not be part of the scale options.